src/Resources/contao/modules/ModuleLogin.php

```diff
@@ -83,72 +72,81 @@ protected function compile()
 		/** @var RouterInterface $router */
 		$router = \System::getContainer()->get('router');
 
-		/** @var TokenInterface $token */
-		$token = \System::getContainer()->get('security.token_storage')->getToken();
-
-		// Do not redirect if authentication is successful
-		if ($token !== null && $token->getUser() instanceof FrontendUser && $token->isAuthenticated())
+		if (\System::getContainer()->get('contao.security.token_checker')->hasFrontendUser())
 		{
+			/** @var PageModel $objPage */
+			global $objPage;
+
 			$this->import('FrontendUser', 'User');
 
+			$strRedirect = \Environment::get('base').\Environment::get('request');
+
+			// Redirect to last page visited
+			if ($this->redirectBack && $_SESSION['LAST_PAGE_VISITED'] != '')
+			{
+				$strRedirect = \Environment::get('base').$_SESSION['LAST_PAGE_VISITED'];
+			}
+
+			// Redirect home if the page is protected
+			elseif ($objPage->protected)
+			{
+				$strRedirect = \Environment::get('base');
```

Comment:
I don't think this is correct. You should leave the decision to the "error 403" handler, which will redirect the user to e.g. the login page.

Reply:
The login handler does not know about pages and their protection, and it does not generate links from page IDs or thelike. Technically it would probably be possible, but I think it would be very complicated. This is just the implementation is always was… can we maybe keep that for now and discuss whether to implement a more advanced version?

Reply:
Just remove the `$objPage->protected` check that you have added in your PR. It worked fine before.

Reply:
Uhm, I added that because it existed before? https://github.com/contao/core-bundle/blob/4.4/src/Resources/contao/modules/ModuleLogin.php#L142

Reply:
Sorry, I missed this. My bad.